src/main/java/com/strandls/document/service/Impl/DocumentServiceImpl.java

```diff
@@ -1487,4 +1504,169 @@ public Boolean documentDownloadLog(HttpServletRequest request, DownloadLogData d
 
 	}
 
+	@SuppressWarnings("deprecation")
+	@Override
+	public GNFinderResponseMap parsePdfWithGNFinder(String filePath, Long documentId) {
+
+		Properties properties = PropertyFileUtil.fetchProperty("config.properties");
+		String serverUrl = properties.getProperty("serverUrl");
+
+		GNFinderResponseMap gnfinderresponse = null;
+		String basePath = properties.getProperty("baseDocPath");
+		String completeFileUrl = serverUrl + basePath + filePath;
+
+		URIBuilder builder = new URIBuilder();
+		builder.setScheme("http").setHost("localhost:3006").setPath("/parse").setParameter("file", completeFileUrl);
+
+		URI uri = null;
+		try {
+			uri = builder.build();
+			HttpGet request = new HttpGet(uri);
+
+			try (CloseableHttpResponse response = httpClient.execute(request)) {
+
+				HttpEntity entity = response.getEntity();
+
+				if (entity != null) {
+					String result = EntityUtils.toString(entity);
+					gnfinderresponse = objectMapper.readValue(result, GNFinderResponseMap.class);
+					gnfinderresponse = MicroServicesUtils.fetchSpeciesDetails(gnfinderresponse, esService);
+					gnfinderresponse = mergeCommonObjects(gnfinderresponse);
+
+					if (documentId != null) {
+						List<GnFinderResponseNames> unsortedNames = gnfinderresponse.getNames();
+						List<GnFinderResponseNames> sortedNames = sortScientificNamesOnFrequency(unsortedNames);
+						saveScientificNamesInTable(sortedNames, documentId);
+					}
+
+				}
+
+			} catch (Exception e) {
+				logger.error(e.getMessage());
+			}
+
+		} catch (URISyntaxException e1) {
+			logger.error(e1.getMessage());
+		}
+
+		return gnfinderresponse;
+	}
+
+	private List<GnFinderResponseNames> sortScientificNamesOnFrequency(List<GnFinderResponseNames> names) {
+		Collections.sort(names, new Comparator<GnFinderResponseNames>() {
+			public int compare(GnFinderResponseNames name1, GnFinderResponseNames name2) {
+				return ((Integer) name2.getFrequency()).compareTo(name1.getFrequency());
+			}
+		});
+
+		return names;
+	}
+
+	private void saveScientificNamesInTable(List<GnFinderResponseNames> names, Long documentId) {
+		int order = 1;
+		for (GnFinderResponseNames name : names) {
+			DocSciName nameTosave = new DocSciName();
+			nameTosave.setCanonicalForm(name.getName());
```

Comment:
use a constructor ,instead of setter methods

Reply:
*Discussion needed, Hence no changes made

Reply:
Corrected.